processor/src/main/java/org/mapstruct/ap/internal/processor/MethodRetrievalProcessor.java

```diff
@@ -507,19 +509,91 @@ private boolean checkParameterAndReturnType(ExecutableElement method, List<Param
             if ( !mappings.containsKey( mappingAnnotation.target() ) ) {
                 mappings.put( mappingAnnotation.target(), new ArrayList<>() );
             }
-            Mapping mapping = Mapping.fromMappingPrism( mappingAnnotation, method, messager, typeUtils );
-            if ( mapping != null ) {
-                mappings.get( mappingAnnotation.target() ).add( mapping );
+
+            if (mappingAnnotation.source().endsWith( ".*" )) {
```

Comment:
The logic of analysing source reference / target reference is in the classes `SourceReference` and `TargetReference`.. Looking from a separation of concerns, I've the feeling this is not the proper place to put this kind of analysis.